dgraph/cmd/alpha/run.go

```diff
@@ -651,6 +651,7 @@ func run() {
 
 	if keys.AclKey != nil {
 		opts.HmacSecret = keys.AclKey
+		opts.UsePublicKey = keys.UsePublicKey
```

Comment:
I am thinking whether it would make more sense to take a more direct approach here. Usually, any JWT supported algorithm will either use a public-private key pair (RSA, ECDSA) or just one key (HMAC). We can use existing `secret-file` flag for HMAC and new flags for asymmetric case such as, "alg" flag for algorithm, "public-key-file" and "private-key-file" for the asymmetric key pair. The default algorithm will be HS256 unless specified. This would be backward compatible too.

Reply:
I just realized that we can derive the public key from the private key. We can skip taking public key as an argument altogether. It's possible you are already doing that, I am not so sure yet.

Reply:
Done